probe-rs/src/flashing/flasher.rs

```diff
@@ -112,48 +112,48 @@ impl<'session> Flasher<'session> {
             .map_err(FlashError::Core)?;
 
         // TODO: Halt & reset target.
-        log::debug!("Halting core {}", self.core_index);
+        tracing::debug!("Halting core {}", self.core_index);
         let cpu_info = core
             .halt(Duration::from_millis(100))
             .map_err(FlashError::Core)?;
-        log::debug!("PC = 0x{:08x}", cpu_info.pc);
-        log::debug!("Reset and halt");
+        tracing::debug!("PC = 0x{:08x}", cpu_info.pc);
+        tracing::debug!("Reset and halt");
         core.reset_and_halt(Duration::from_millis(500))
             .map_err(FlashError::Core)?;
 
         // TODO: Possible special preparation of the target such as enabling faster clocks for the flash e.g.
 
         // Load flash algorithm code into target RAM.
-        log::debug!(
-            "Loading algorithm into RAM at address 0x{:08x}",
-            algo.load_address
-        );
+        let span = tracing::debug_span!("Loading algorithm into RAM", address = algo.load_address)
+            .entered();
 
         core.write_32(algo.load_address as u64, algo.instructions.as_slice())
             .map_err(FlashError::Core)?;
 
+        let _span = span.exit();
```

Comment:
Why not drop here like the other places?

Reply:
🤷 No idea, adapted it for consistency.